src/function.js

```diff
@@ -127,7 +127,7 @@ Sk.builtin.func = Sk.abstr.buildNativeClass("function", {
         },
         __doc__: {
             $get() {
-                return new Sk.builtin.str(this.$doc);
+                return this.$doc;
             },
```

Comment:
we should add a `$set`

```javascript
$set(v) {
    this.$doc = v; // can be an arbitrary python object
}
```

And perhaps a test.
You should be able to set the `__doc__` in python for any function.
Note that the `__doc__` of a method is readonly

```python
class A: 
    def foo(self): 
        "docstring" 
                                                                
A.foo.__doc__ = 'bar'                                                  
A.foo.__doc__                                                          
#'bar'

a = A()                                                                

a.foo.__doc__ = 'baz' 
# AttributeError: attribute '__doc__' of 'method' objects is not writable
# this is already implemented

A.foo.__doc__ = 1 # can be arbitrary
```

Reply:
Good idea.  I've added the code you suggest, and also some tests based on your examples.